runtime/default.go

```diff
@@ -204,19 +78,53 @@ func (r *runtime) run() {
 			if service.Running() {
 				continue
 			}
-
 			// TODO: check service error
-			log.Debugf("Starting %s", service.Name)
+			log.Debugf("Runtime starting service %s", service.Name)
 			if err := service.Start(); err != nil {
-				log.Debugf("Runtime error starting %s: %v", service.Name, err)
+				log.Debugf("Runtime error starting service %s: %v", service.Name, err)
+			}
+		case event := <-events:
+			log.Debugf("Runtime received notification event: %v", event)
+			// NOTE: we only handle Update events for now
+			switch event.Type {
+			case Update:
+				// parse returned response to timestamp
+				buildTime, err := time.Parse(time.RFC3339, event.Version)
+				if err != nil {
+					log.Debugf("Runtime error parsing build time: %v", err)
+					continue
+				}
+				r.Lock()
+				for _, service := range r.services {
+					muBuild, err := time.Parse(time.RFC3339, service.Version)
+					if err != nil {
+						log.Debugf("Runtime could not parse %s service build: %v", service.Name, err)
+						continue
+					}
+					if buildTime.After(muBuild) {
+						if err := r.Update(service.Service); err != nil {
+							log.Debugf("Runtime error updating service %s: %v", service.Name, err)
+							continue
+						}
+						service.Version = event.Version
+					}
+				}
+				r.Unlock()
+			}
+		case <-r.closed:
+			log.Debugf("Runtime stopped. Attempting to stop all services.")
+			for name, service := range r.services {
+				// TODO: handle this error
+				if err := r.Delete(service.Service); err != nil {
```

Comment:
Let's not stop the services right now

Reply:
This is a local runtime.  Delete stops them. I understand why we dont want to do this in K8s runtime right now, but being able to stop multiple services when stopping local runtime seems reasonable. Otherwise we're gonna end up chasing child process PIDs manually?

Reply:
The processes are children of this runtime and since Go can't fork they'll all die. But also micro/cmd called runtime.Delete explicitly. I think the caller who creates should also delete.